### PR TITLE
Components

### PR DESCRIPTION
**Pull request by M0rrighan**
Added class based components:

- App
- Calculator
- Buttons
- Button
- Display

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,69 @@
...
-This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
-
-### Deployment
-
-This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)
-
-### `npm run build` fails to minify
-
-This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
+This project is [MIT](./MIT.md) licensed.
```
Comment (ahmetbozaci) [inline README.md]: - [ ] If you add a license broken link will fix it. Please fix the broken link in the license section.

File: src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable react/prefer-stateless-function */
```
Comment (ahmetbozaci) [inline src/components/App.js:1]: - [ ] You shouldn't disable linters. Please delete it and fix the error.
For stateless function,you can check this [link](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md)

File: src/components/Button.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-useless-constructor */
+/* eslint-disable react/prefer-stateless-function */
```
Comment (ahmetbozaci) [inline src/components/Button.js:2]: - [ ] You shouldn't disable linters. Please delete it and fix the error.
For stateless function,you can check this [link](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md)

File: src/components/Buttons.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable react/prefer-stateless-function */
```
Comment (ahmetbozaci) [inline src/components/Buttons.js:1]: - [ ] You shouldn't disable linters. Please delete it and fix the error.
For stateless function,you can check this [link](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md)

File: src/components/Calculator.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable react/prefer-stateless-function */
```
Comment (ahmetbozaci) [inline src/components/Calculator.js:1]: - [ ] You shouldn't disable linters. Please delete it and fix the error.
For stateless function,you can check this [link](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md)

File: src/components/Display.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-useless-constructor */
+/* eslint-disable react/prefer-stateless-function */
```
Comment (ahmetbozaci) [inline src/components/Display.js:2]: - [ ] You shouldn't disable linters. Please delete it and fix the error.
For stateless function,you can check this [link](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md)

File: src/index.css [hunk 7]
```diff
@@ -1,27 +1,80 @@
...
+  padding-right: 0.5em;
+}
+
+.buttons {
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr 1fr;
+}
+
+.btn {
+  display: flex;
```
Comment (ahmetbozaci) [inline src/index.css:40]: ### OPTIONAL
- [ ] For DRY CSS you should combine duplicate properties. 
Example, you can create class `.d-flex { display: flex}` and use this class in HTML file.

- [More information about DRY principles](https://www.freecodecamp.org/news/7-important-tips-for-writing-better-css/) 

- [Html and css best practices in curriculum](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md)

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️
hello 
@M0rrighan  ✋🏼 
#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯

- Nice to use different components for buttons. 
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

almost

___

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_


_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mentioning me as [ahmetbozaci](https://github.com/ahmetbozaci), I will receive a notification and will respond._
___
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: src/components/App.js (hunk 2 above)
Comment (M0rrighan) [reply src/components/App.js:1]: Refactoring to functional components is a separate milestone of this project.

File: src/components/Button.js (hunk 3 above)
Comment (M0rrighan) [reply src/components/Button.js:2]: Refactoring to functional components is a separate milestone of this project.

File: src/components/Buttons.js (hunk 4 above)
Comment (M0rrighan) [reply src/components/Buttons.js:1]: Refactoring to functional components is a separate milestone of this project.

File: src/components/Calculator.js (hunk 5 above)
Comment (M0rrighan) [reply src/components/Calculator.js:1]: Refactoring to functional components is a separate milestone of this project.

File: src/components/Display.js (hunk 6 above)
Comment (M0rrighan) [reply src/components/Display.js:2]: Refactoring to functional components is a separate milestone of this project.

**Review by er-contreras — APPROVED** [review]
Hi @M0rrighan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- I see that you did most of the work, well done!

Felix

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional 

- [ ] It's important to learn how to work without disabling Linter. Further up try to build your app without disabling Linters.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Display.js (hunk 6 above)
Comment (ahmetbozaci) [reply src/components/Display.js:2]: @M0rrighan It is not a functional component. You can use still class. If you can check the link that ı send to you it is show how to fix this error.

Comment (ahmetbozaci) [reply src/components/Display.js:2]:
```
import React, { PureComponent } from 'react';

class Calculator extends PureComponent { // or React.PureComponent

...

}
```

**Pull request merged**